shared/metrics/src/main/java/io/pravega/shared/metrics/StatsLoggerProxy.java

```diff
@@ -65,6 +65,16 @@ public Gauge registerGauge(String name, Supplier<Number> supplier, String... tag
                 (metric, proxyName, c) -> new GaugeProxy(metric, proxyName, c), tags);
     }
 
+    @Override
+    public void deleteGauge(String name, String... tags) {
+        MetricsNames.MetricKey keys = metricKey(name, tags);
+        Metric existingGauge = this.gauges.get(keys.getCacheKey());
```

Comment:
Replace `get` with `remove`

Reply:
Will do in the separate PR focusing on segments count